amd/src/courseformat/tabtree.js

```diff
@@ -0,0 +1,176 @@
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+import {BaseComponent} from 'core/reactive';
+import {getCurrentCourseEditor} from 'core_courseformat/courseeditor';
+import Templates from 'core/templates';
+
+
+/**
+ * Course section tabs updater.
+ *
+ * @module     format_multitopic/courseformat/tabtree
+ * @class      format_multitopic/courseformat/tabtree
+ * @copyright  2022 Jeremy FitzPatrick and Te Wānanga o Aotearoa
+ * @copyright  based on work by 2021 Ferran Recio <ferran@moodle.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+export default class Component extends BaseComponent {
+
+    /**
+     * Constructor hook.
+     */
+    create() {
+        // Optional component name for debugging.
+        this.name = 'coursetabs';
+        // Default query selectors.
+        this.selectors = {
+            TAB: `ul:first-of-type li`,
+            CHILDTAB: `ul:nth-child(2) li`
+        };
+        // Default classes
+        this.classes = {
+            ACTIVETAB: 'active'
+        };
+        // Objects to keep tabs on the tabs
+        this.tabs = {};
+        this.childtabs = {};
+        this.activetab = 0;
```

Comment:
We need to find the active tab at each level.
```suggestion
        this.activetab = [null, null];
```